tests.py

```diff
@@ -1,32 +1,70 @@
+import json
+import collections
+from mock import patch
+from datetime import datetime
 from cStringIO import StringIO
-from wrapplog import Logger, start_logging
 
+from wrapplog import Logger
 
 
 class TestWrappObserver(object):
-    def setup(self):
+    @patch('wrapplog._timestamp')
+    def setup(self, timestamp_mock):
+        self.timestamp = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+        timestamp_mock.return_value = self.timestamp
         self.out = StringIO()
-        start_logging(self.out)
-        self.log = Logger()
+        self.msg = 'Hello'
+        self.service = "api"
+        self.host = "host-01"
+        self.namespace = 'tests'
+        self.metric_value = 10
+        self.event_data = {"user": {"name": "jude", "id": 1}}
+        self.log = Logger(self.out, service=self.service, host=self.host)
+
+    def _generate_output(self, level):
+        res = collections.OrderedDict()
+        res['level'] = level
+        if level in ['event', 'metric']:
+            res[level] = self.msg
+        else:
+            res['msg'] = self.msg
+
+        if level == 'event':
+            res['data'] = self.event_data
+        res['host'] = self.host
+        res['namespace'] = self.namespace
+        res['service'] = self.service
+        res['timestamp'] = self.timestamp
+        if level == 'metric':
+            res['value'] = self.metric_value
+        return '%s %s\n' % (level.upper(), json.dumps(res))
 
     def test_debug(self):
-        self.log.debug('Hello!')
-        self.assert_output('DEBUG {"level": "debug", "msg": "Hello!", "namespace": "tests"}\n')
+        self.log.debug(self.msg)
+        self.assert_output(self._generate_output('debug'))
 
     def test_info(self):
-        self.log.info('Hello!')
-        self.assert_output('INFO {"level": "info", "msg": "Hello!", "namespace": "tests"}\n')
+        self.log.info(self.msg)
+        self.assert_output(self._generate_output('info'))
 
     def test_warning(self):
-        self.log.warning('Hello!')
-        self.assert_output('WARNING {"level": "warning", "msg": "Hello!", "namespace": "tests"}\n')
+        self.log.warning(self.msg)
+        self.assert_output(self._generate_output('warning'))
 
     def test_error(self):
-        self.log.error('Hello!')
-        self.assert_output('ERROR {"level": "error", "msg": "Hello!", "namespace": "tests"}\n')
+        self.log.error(self.msg)
+        self.assert_output(self._generate_output('error'))
+
+    def test_event(self):
+        self.log.event(self.msg, self.event_data)
+        self.assert_output(self._generate_output('event'))
+
+    def test_metric(self):
+        self.log.metric(self.msg, self.metric_value)
+        self.assert_output(self._generate_output('metric'))
```

Comment:
I don't like old or new tests :) Both are sensitive to formatting rules (that might or might not be the same). But I guess it is OK. If test pass we know format is ok, if they fail we are not sure they are broken .